compiler/rustc_span/src/edit_distance.rs

```diff
@@ -198,12 +199,27 @@ fn find_best_match_for_name_impl(
         } {
             Some(0) => return Some(*c),
             Some(d) => {
-                dist = d - 1;
-                best = Some(*c);
+                if use_substring_score {
+                    dist = d;
+                    next_candidates.push(*c);
+                    best = Some(*c);
+                } else {
+                    dist = d - 1;
+                    best = Some(*c);
+                }
             }
             None => {}
         }
     }
+
+    if next_candidates.len() > 1 {
```

Comment:
Could you add a comment stating what happens here?
Does this need an `debug_assert!(use_substring_score)` to emphasize that only happens in that case?

Reply:
yes, added the assertion in code.

Reply:
r=me with a comment explaining this branch. Example (to be adapted)
```suggestion
    //  We have a tie among several candidates, try to select the best among them ignoring substrings.
    if next_candidates.len() > 1 {
```